### PR TITLE
Bump up version to `0.18.1`

### PR DESCRIPTION
**Pull request by noklam**
Signed-off-by: noklam <nok.lam.chan@quantumblack.com>

## Description

Release a new minor version of Kedro 0.18.1.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1519"></a>

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -208,7 +208,6 @@
     "https://docs.delta.io/latest/delta-update.html#language-python",
     "https://github.com/kedro-org/kedro/blob/main/kedro_technical_charter.pdf",
 
-    # Number section removal, temporarily ignore until 0.18.0 release
```
Comment (MerelTheisenQB) [inline docs/conf.py:211]: I think actually this comment was referring to all the links below, so we should be able to now remove those from the ignored links.

**Review by SajidAlamQB — COMMENTED** [review]
`kedro/docs/source/development/commands_references.md` L:117 also needs changing.

Edit: Nevermind you got it. 👍

**Review by SajidAlamQB — APPROVED** [review]
Awesome thank you! 💥

File: RELEASE.md [hunk 2]
```diff
@@ -15,12 +15,12 @@
 * Added a new hook `after_context_created` that passes the `KedroContext` instance as `context`.
 * Added more detail to YAML ParserError error message.
 * Added option to `SparkDataSet` to specify a `schema` load argument that allows for supplying a user-defined schema as opposed to relying on the schema inference of Spark.
+* Introduced `after_command_run` CLI hook.
```
Comment (AntonyMilneQB) [inline RELEASE.md:18]:
```suggestion
* Added a new hook `after_context_created` that passes the `KedroContext` instance as `context`.
* Added a new CLI hook `after_command_run`.
* Added more detail to YAML ParserError error message.
* Added option to `SparkDataSet` to specify a `schema` load argument that allows for supplying a user-defined schema as opposed to relying on the schema inference of Spark.
```

**Review by AntonyMilneQB — APPROVED** [review]
Small proposed tidying to RELEASE.md but other than that looks good!

I think `* The Kedro package no longer contains a built version of the Kedro documentation significantly reducing the package size.` counts as a major improvement also 🙂

File: static/jsonschema/kedro-catalog-0.18.json [hunk 3]
```diff
@@ -97,7 +97,7 @@
               },
               "checkpoint": {
                 "pattern": "object",
-                "description": "Optional checkpoint configuration. Accepts a dictionary\nwith the corresponding dataset definition including ``filepath``\n(unlike ``dataset`` argument). Checkpoint configuration is\ndescribed here:\nhttps://kedro.readthedocs.io/en/0.18.0/data/kedro_io.html#checkpoint-configuration\nCredentials for the checkpoint can be explicitly specified\nin this configuration."
+                "description": "Optional checkpoint configuration. Accepts a dictionary\nwith the corresponding dataset definition including ``filepath``\n(unlike ``dataset`` argument). Checkpoint configuration is\ndescribed here:\nhttps://kedro.readthedocs.io/en/0.18.1/data/kedro_io.html#checkpoint-configuration\nCredentials for the checkpoint can be explicitly specified\nin this configuration."
```
[the remark this answers is not in the document]
Comment (noklam) [reply static/jsonschema/kedro-catalog-0.18.json:100]: Noted, thanks!

**Pull request merged**